test/performance/scale_from_zero_test.go

```diff
@@ -57,21 +57,21 @@ func runScaleFromZero(idx int, t *testing.T, clients *test.Clients, ro *test.Res
 		test.DeploymentScaledToZeroFunc,
 		"DeploymentScaledToZero",
 		test.ServingNamespace,
-		2*time.Minute); err != nil {
+		3*time.Minute); err != nil {
 		m := fmt.Sprintf("%d: failed waiting for deployment to scale to zero: %v", idx, err)
 		t.Log(m)
 		return 0, errors.New(m)
 	}
 
 	start := time.Now()
-	t.Log("Waiting for endpoint to serve request")
-	if _, err := pkgTest.WaitForEndpointState(
+	t.Logf("%d: waiting for endpoint to serve request", idx)
+	if _, err := pkgTest.WaitForEndpointStateWithTimeout(
 		clients.KubeClient,
 		t.Logf,
 		domain,
 		test.RetryingRouteInconsistency(pkgTest.MatchesAllOf(pkgTest.IsStatusOK(), pkgTest.MatchesBody(helloWorldExpectedOutput))),
 		"HelloWorldServesText",
-		test.ServingFlags.ResolvableDomain); err != nil {
+		test.ServingFlags.ResolvableDomain, 10*time.Minute); err != nil {
```

Comment:
why not use the timeout constant here?

Reply:
Existing one, or to define a new one?

Reply:
Define a new one unless you think they should be set the same then make the other one public with a slightly more descriptive name.

Reply:
No, those are orthogonal.

Reply:
As in, they are equal here, but they might change independently.

Reply:
Added the constant.